components/screens/ChangePassword.tsx

```diff
@@ -0,0 +1,218 @@
+import React, { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { useForm } from "react-hook-form";
+import { twMerge } from "tailwind-merge";
+import Header from "@/components/GeneralHeader";
+import useKeyring from "@/hooks/useKeyring.ts";
+
+interface PasswordFormData {
+  currentPassword: string;
+  newPassword: string;
+  confirmPassword: string;
+}
+
+export default function ChangePassword() {
+  const { keyringCheckPassword, keyringChangePassword } = useKeyring();
+  const navigate = useNavigate();
+  const [showCurrentPassword, setShowCurrentPassword] = useState(false);
+  const [showNewPassword, setShowNewPassword] = useState(false);
+  const [showConfirmPassword, setShowConfirmPassword] = useState(false);
+  const [isMismatchShown, setIsMismatchShown] = useState(false);
+
+  const {
+    register,
+    handleSubmit,
+    watch,
+    formState: { isValid, errors },
+    setError,
+    clearErrors,
+  } = useForm<PasswordFormData>({
+    mode: "onChange",
+  });
+
+  const { currentPassword, newPassword, confirmPassword } = watch();
+  const passwordMatch = newPassword === confirmPassword;
+  const isDisabled =
+    !currentPassword || !newPassword || !passwordMatch || !isValid;
+
+  const onSubmit = handleSubmit(async (data) => {
+    await keyringChangePassword({
+      currentPassword: data.currentPassword,
+      newPassword: data.newPassword,
+    });
+
+    navigate("/settings");
+  });
+
+  useEffect(() => {
+    let timeout: NodeJS.Timeout | undefined;
+
+    const hasMismatch = !passwordMatch && !!confirmPassword;
+
+    if (!hasMismatch) {
+      setIsMismatchShown(hasMismatch);
+    } else {
+      timeout = setTimeout(() => {
+        setIsMismatchShown(hasMismatch);
+      }, 600);
+    }
+    return () => clearTimeout(timeout);
+  }, [newPassword, confirmPassword]);
+
+  useEffect(() => {
+    let timeout: NodeJS.Timeout | undefined;
+
+    const validatePassword = async () => {
+      const { success } = await keyringCheckPassword({
+        password: currentPassword,
+      });
```

Comment:
I think we can skip the validation of empty password.
![Screenshot 2025-04-01 at 11 42 37 AM](https://github.com/user-attachments/assets/f27ee09b-84ee-4320-9021-3b148193fe08)

Reply:
2a996c64f4a12469b277fabd01af5e2199f91c2e

I will also make sure Nicole approves the debounce behavior because it's not the best thing right now